server/channels/app/platform/web_hub.go

```diff
@@ -492,14 +494,19 @@ func (h *Hub) Start() {
 				if metrics := h.platform.metricsIFace; metrics != nil {
 					metrics.DecrementWebSocketBroadcastBufferSize(strconv.Itoa(h.connectionIndex), 1)
 				}
+
+				// Remove the broadcast hook information before precomputing the JSON so that those aren't included in it
+				msg, broadcastHooks, broadcastHookArgs := msg.WithoutBroadcastHooks()
+
 				msg = msg.PrecomputeJSON()
+
 				broadcast := func(webConn *WebConn) {
 					if !connIndex.Has(webConn) {
 						return
 					}
 					if webConn.ShouldSendEvent(msg) {
 						select {
-						case webConn.send <- msg:
+						case webConn.send <- h.runBroadcastHooks(msg, webConn, broadcastHooks, broadcastHookArgs):
```

Comment:
Is it necessary to run the hooks while sending? Wondering whether hooks should run the hooks here or should we run those where it's been received?

Reply:
I think this is probably the best spot for it for now. I do agree that running it in the `WebConn` does feel right because we want these to run right before an event is sent to the client, but that also requires registering each broadcast hook with every `WebConn` which feels like we're passing that information around a lot. Because of that, I think it's fine for the `WebHub` to be running them. Only 1/5 though.

Reply:
I see, thanks for the explanation.